File: android/src/main/java/expo/modules/nativesettingselapsedtime/NativeSettingsElapsedTimeModule.kt
package expo.modules.nativesettingselapsedtime

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import android.os.SystemClock

class NativeSettingsElapsedTimeModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("NativeSettingsElapsedTime")

        Function("getElapsedTimeMillis") {
            return@Function SystemClock.elapsedRealtime()
        }
    }
}

File: ios/NativeSettingsElapsedTimeModule.swift
import ExpoModulesCore

public class NativeSettingsElapsedTimeModule: Module {
    public func definition() -> ModuleDefinition {
        Name("NativeSettingsElapsedTime")

        Function("getElapsedTimeMillis") { () -> Double in
            let nanoseconds = clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW)
            return Double(nanoseconds) / 1_000_000.0
        }
    }
}